src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,636 @@
+import com.sun.jersey.api.client.ClientRequest;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.config.ClientConfig;
+import com.sun.jersey.api.client.config.DefaultClientConfig;
+
+import edu.cornell.kfs.fp.CuFPConstants;
+import edu.cornell.kfs.fp.businessobject.AmazonBillingCostCenterDTO;
+import edu.cornell.kfs.fp.businessobject.AmazonBillingDistributionOfIncomeTransactionDTO;
+import edu.cornell.kfs.fp.document.CuDistributionOfIncomeAndExpenseDocument;
+import edu.cornell.kfs.fp.document.service.impl.CULegacyTravelServiceImpl;
+import edu.cornell.kfs.fp.service.AmazonWebServicesBillingService;
+import edu.cornell.kfs.fp.xmlObjects.AmazonAccountDetail;
+import edu.cornell.kfs.fp.xmlObjects.AmazonAccountDetailContainer;
+import edu.cornell.kfs.sys.CUKFSConstants;
+
+public class AmazonWebServicesBillingServiceImpl implements AmazonWebServicesBillingService, Serializable {
+
+    private static final long serialVersionUID = 7430710204404759511L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(AmazonWebServicesBillingServiceImpl.class);
+    
+    private String awsURL;
+    private String awsToken;
+    private String billingPeriodParameterValue;
+    private String defaultDocumentDescription;
+    private String transactionObjectCode;
+    private String transactionFromAccountNumber;
+    private String helpNoteText;
+
+    protected ParameterService parameterService;
+    protected DocumentService documentService;
+    protected ChartService chartService;
+    protected AccountService accountService;
+    protected SubAccountService subAccountService;
+    protected BusinessObjectService businessObjectService;
+    protected ObjectCodeService objectCodeService;
+    protected SubObjectCodeService subObjectCodeService;
+    protected ProjectCodeService projectCodeService;
+
+    @Override
+    public void generateDistributionOfIncomeDocumentsFromAWSService() {
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService(): AWS URL: " + getAwsURL());
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService(): Processing Year: " + findProcessYear() + "  Month: " + findProcessMonthNumber());
+        
+        String jsonResults = buildJsonOutput();
+        LOG.debug("JSON results: " + jsonResults);
+
+        List<AmazonAccountDetail> amazonAccountDetails = buildAmazonAcountListFromJson(jsonResults);
+        LOG.info("generateDistributionOfIncomeDocumentsFromAWSService() Number of amazon account details received: " + amazonAccountDetails.size());
+        
+        boolean valid = true;
+        
+        if (amazonAccountDetails.size() == 0) {
+            LOG.error("generateDistributionOfIncomeDocumentsFromAWSService() No amazon account detail objects parsed from the webservice.");
+            valid =  false;
+        }
+        
+        valid = valid && processAmazonAccountDetails(amazonAccountDetails);
```

Comment:
Valid is only false when there aren't any records in amazonAccountDetails, so there are no records to process anyway